java-checks/src/main/java/org/sonar/java/symexecengine/DataFlowVisitor.java

```diff
@@ -69,8 +93,9 @@ public void visitVariable(VariableTree tree) {
   public void visitAssignmentExpression(AssignmentExpressionTree tree) {
     super.visitAssignmentExpression(tree);
     Symbol symbol = getSymbol(tree.variable());
-    if(symbol != null && isSymbolRelevant(symbol)) {
+    if (symbol != null) {
       executionState.createValueForSymbol(symbol, tree.expression());
+      check.onAssignment(executionState, tree, symbol, tree.expression());
```

Comment:
![New](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/new.png) ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Method 'SymbolicExecutionCheck.onAssignment(...)' is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)